src/main/java/org/terracotta/statistics/SuppliedValueStatistic.java

```diff
@@ -77,11 +96,7 @@ public T value() {
         long now = System.nanoTime();
         long exp = expiration.get();
         if (now >= exp && expiration.compareAndSet(exp, now + delayNs)) {
-          T current = memoized.get();
-          T newValue = valueStatistic.value();
-          if (memoized.compareAndSet(current, newValue)) {
```

Comment:
in the corner case where thread 1 starts updating, and caching delay < time to compute stat, then thread 2 could come and start updating the value also. In this case, the second `compareAndSet`  was making sure only the thread 1 is able to set his new value, which is dumb because if some other thread starts updating the value, we would like to see the new update also ;-)

Reply:
note: the way we collect stats from the management layer, it is single-threaded anyway, so this won't even happen...